File: src/main/Tracking_food/Sound.cpp
#include "Sound.h"
#include <iostream>
#include <cstdlib>
#include <string>

Sound::Sound(){}
Sound::~Sound(){}
void Sound::play_sound(std::string dish){
    // cmd 창에서 실행된다고 생각
    // wav 파일만. mp3 파일 x
    if(dish == "none"){
        system("canberra-gtk-play -f ~/Tracking-Food/data/sound/더 이동해주세요.wav");
        return;
    }
    else if(dish =="scan_start"){
        system("canberra-gtk-play -f ~/Tracking-Food/data/sound/스캔중.wav");s
        return;
    }
    else if(dish=="scan_end"){
        system("canberra-gtk-play -f ~/Tracking-Food/data/sound/스캔완료.wav");
        return;
    }
    else{
        system(("canberra-gtk-play -f ~/Tracking-Food/data/sound/"+dish+".wav").c_str());
        return;
    }

    // if(dish == "계란말이"){
    //     system("canberra-gtk-play -f ~/Tracking-Food/data/sound/계란말이.wav");
    //     return;
    // }

    // if(dish == "김치"){
    //     system("canberra-gtk-play -f ~/Tracking-Food/data/sound/김치.wav");
    //     return;
    // } 

    // if(dish == "김치국"){
    //     system("canberra-gtk-play -f ~/Tracking-Food/data/sound/김치국.wav");
    //     return;
    // }

    // if(dish == "된장국"){
    //     system("canberra-gtk-play -f ~/Tracking-Food/data/sound/된장국.wav");
    //     return;
    // }

    // if(dish == "두부조림"){
    //     system("canberra-gtk-play -f ~/Tracking-Food/data/sound/두부조림.wav");
    //     return;
    // }
    
    // if(dish == "멸치조림"){
    //     system("canberra-gtk-play -f ~/Tracking-Food/data/sound/멸치조림.wav");
    //     return;
    // }
    // if(dish == "미역국"){
    //     system("canberra-gtk-play -f ~/Tracking-Food/data/sound/미역국.wav");
    //     return;
    // }
    // if(dish == "북엇국"){
    //     system("canberra-gtk-play -f ~/Tracking-Food/data/sound/북엇국.wav");
    //     return;
    // }
    // if(dish == "불고기"){
    //     system("canberra-gtk-play -f ~/Tracking-Food/data/sound/불고기.wav");
    //     return;
    // }
    // if(dish == "시금치"){
    //     system("canberra-gtk-play -f ~/Tracking-Food/data/sound/시금치.wav");
    //     return;
    // }
    // if(dish == "우엉조림"){
    //     system("canberra-gtk-play -f ~/Tracking-Food/data/sound/우엉조림.wav");
    //     return;
    // }
    // if(dish == "육개장"){
    //     system("canberra-gtk-play -f ~/Tracking-Food/data/sound/육개장.wav");
    //     return;
    // }
    
    // if(dish == "장조림"){
    //     system("canberra-gtk-play -f ~/Tracking-Food/data/sound/장조림.wav");
    //     return;
    // }

    // if(dish == "현미밥"){
    //     system("canberra-gtk-play -f ~/Tracking-Food/data/sound/현미밥.wav");
    //     return;
    // }
    // if(dish == "흰밥"){
    //     system("canberra-gtk-play -f ~/Tracking-Food/data/sound/흰밥.wav");
    //     return;
    // }




}

File: src/sound.py
#!/usr/local/bin/python3
# coding=utf-8

# Copyright (c) 2020 KOBOT kobot9@naver.com.
# All rights reserved.

# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:

# The above copyright notice and this permission notice shall be included in all
# copies or substantial portions of the Software.

# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
# SOFTWARE.


import os
import multiprocessing


# "play sound in cmd" not using playsound module
class Sound:
    def __init__(self,answer):

        p = multiprocessing.Process(name="SubProcess", target=self.play_sound, args=(answer,))
        p.start()


    def play_sound(self, dish_name):
        if dish_name == "none": # 아무 것도 들어오지 않은 경우
            os.system("canberra-gtk-play -f ../data/sound/더이동해주세요.wav")
            return
        elif dish_name == "scan_start": # 스캔 시작 알림
            os.system("canberra-gtk-play -f ../data/sound/스캔중.wav")
            return

        elif dish_name =="scan_end": # 스캔 완료 알림
            os.system("canberra-gtk-play -f ../data/sound/스캔완료.wav")
            return
        elif dish_name == "scan_find": # 스캔 시작 알림
            os.system("canberra-gtk-play -f ../data/sound/음식이보입니다.wav")
            return
        else: # 반찬 이름이 들어울 경우
            os.system("canberra-gtk-play -f ../data/sound/" + str(dish_name) + ".wav")
            return
        

